dev/yarn.ts

```diff
@@ -1,30 +1,30 @@
+const yarnGenerators = {
+  getScripts: {
+    script: "cat package.json",
+    postProcess: function (output) {
+      if (output.trim() == "") {
+        return [];
+      }
+      try {
+        const packageContent = JSON.parse(output);
+        const scripts = packageContent["scripts"];
+        if (scripts) {
+          return Object.keys(scripts).map((scriptName) => ({
+            name: scriptName,
+            icon: "https://yarnpkg.com/favicon-32x32.png",
+          }));
+        }
+      } catch (e) {}
+
+      return [];
+    },
+  } as Fig.Generator,
+};
```

Comment:
We shouldn't use type assertion, rather infer the type at the declaration level:
```suggestion
const yarnGenerators: Record<string, Fig.Generator> = {
  getScripts: {
    script: "cat package.json",
    postProcess: function (output) {
      if (output.trim() == "") {
        return [];
      }
      try {
        const packageContent = JSON.parse(output);
        const scripts = packageContent["scripts"];
        if (scripts) {
          return Object.keys(scripts).map((scriptName) => ({
            name: scriptName,
            icon: "https://yarnpkg.com/favicon-32x32.png",
          }));
        }
      } catch (e) {}

      return [];
    },
  },
};
```

Reply:
I did this before, but doing that causes autocompletion when typing `yarnGenerators.` not to work. When you use `string` as the key of a record it means it can be any `string` and it will not show you (atleast in vscode) completions for `getScripts`

Reply:
Nevermind, you're right. We could hard write the `getScripts` property like this to get the completion: `const yarnGenerators: { getScripts: Fig.Generator } = {`, but it's not very clean.